### DISCUSSION TITLE
Keep-alive message causes an exception

### DISCUSSION THREAD

**Discussion by BruceCarson** [Marfusios/websocket-client#83] [Ideas]
I'm using a websocket service that uses a single newline character "\n" as a keep-alive message.  However on send, Websocket.Client throws an exception. I receive the keep-alives FROM the server (also "\n") with no issues.

Is there any problem with changing the ValidateInput(string value, string name) to using:
string.IsNullOrEmpty(value)
instead of:
string.IsNullOrWhitespace(value)

If there is no problem, could you change in the next version?

**Comment by Marfusios** [comment]
Hello @BruceCarson , 

seems fine to me, could you provide PR with the change?

**Reply by BruceCarson** [reply]
One of your tests is failing - even without changing code: OnClose_ShouldWorkCorrectly.  I get:
Assert.Equal() Expected: Aborted Actual: Closed
Seems like this test is out-of-date?
I'll change to expecting Closed in the pull request - but it has nothing to do with my change ;)

**Reply by BruceCarson** [reply]
Done.

**Reply by Marfusios** [reply]
Hmm, that's weird, hopefully, it is not a breaking change in a .NET Core

**Comment by BruceCarson** [comment]
Update:
I worked around the issue by doing this:
           var bytes = Encoding.UTF8.GetBytes("\n");
            Client.Send(bytes);
But also submitted a pull request to allow using just: Client.Send("\n");
